tools/githooks/setup.py

```diff
@@ -20,13 +19,17 @@
 FLUTTER_DIR = os.path.join(SRC_ROOT, 'flutter')
 
 
+def IsWindows():
+  os_id = sys.platform
+  return os_id.startswith('win32') or os_id.startswith('cygwin')
+
+
 def Main(argv):
+  git = 'git'
   githooks = os.path.join(FLUTTER_DIR, 'tools', 'githooks')
-  git_candidates = ['git', 'git.sh', 'git.bat']
-  git = next(filter(which, git_candidates), None)
-  if git is None:
-    candidates = "', '".join(git_candidates)
-    raise IOError(f"Looks like GIT is not on the path. Tried '{candidates}'")
+  if IsWindows():
+    git = 'git.bat'
+    githooks = os.path.join(githooks, 'windows')
```

Comment:
This is the line that causes the issue. This line shouldn't be updated before the non-Windows pre-push githook is fixed to work on Windows.

Reply:
The non-Windows pre-push works perfectly on Windows since forever, the issue is with python executable discovery / requirement as the main pre-push is a python script.